packages/app-bridge/src/AppBridgePlatformApp.ts

```diff
@@ -0,0 +1,43 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { ErrorMessageBus, IMessageBus, MessageBus } from './utilities/MessageBus';
+import { generateRandomString, notify, subscribe } from './utilities';
+import { MethodsListKeys, PlatformAppMethods, Topic } from './types';
+import { InitializationError } from './errors/InitializationError';
+import { getQueryParameters } from './utilities/queryParams';
+
+const PUBSUB_CHECKSUM = generateRandomString();
+
+type InitializeEvent = {
+    port: MessagePort;
+};
+
+export class AppBridgePlatformApp {
+    private messageBus: IMessageBus = new ErrorMessageBus();
+    private initialized = false;
+
+    public async initialize() {
```

Comment:
Could probably be a `dispatch({ name: "initializePlatformApp" }): Promise<void>` (`name` field to be challenged by eco-system)

Reply:
Do you mean the entire `initialize()` method would be replaced by this dispatch call? I might want to hop on a call to dig deeper on how to go about it, then :)

Just as an example how we currently figured the usage to be:

```
import "./App.css";
import { AppBridgePlatformApp } from "@frontify/app-bridge";
import { useEffect, useState } from "react";

function App() {
  const [content, setContent] = useState<any>();
  const [appBridge, setAppBridge] = useState<any>();

  useEffect(() => {
    (async () => {
      const appBridge = new AppBridgePlatformApp();
      await appBridge.initialize();
      setAppBridge(appBridge);
    })();
  }, []);

  return (
    <>
      <button
        onClick={async () => {
          const { name } = await appBridge.api({ operation: "currentUser" });
          setContent(name);
        }}
      >
        button
      </button>
      <p>{content}</p>
    </>
  );
}

export default App;

```